packages/engine/errors/2352.md

```diff
@@ -22,3 +22,5 @@ If you really meant to do this, you should cast `{0}` to `unknown` first. For ex
 ```ts twoslash
 const a = "wow" as unknown as string[];
 ```
+
+However, casting a type to `unknown` only to cast it to another more specific type can be dangerous. You should only do this when you are certain that the type you are casting it to matches every possible value at runtime. You may also want to consider why you have to cast the type in the first place, could you improve some other type upstream and avoid casting it altogether?
```

Comment:
I'm torn about this - unsure how much our job should be 'here's what you should do in TS' vs 'here's what this error message means'. I'm edging towards the latter, meaning I don't think we should merge this - despite it being good advice.

Reply:
Well, TypeScript _is_ encoding advice here. 

> If this was intentional, convert the expression to 'unknown' first."

It's arguably more harmful to give this kind of advice that _could_ be the wrong advice without contextualizing it.

With this project, I would also encourage to not structure "translations" exclusively around how TypeScript structures the error / what it chooses to include. Some of the messages are very bad, and this project could also suggest better language to TypeScript itself

Reply:
I think I disagree for now, let's narrow the scope to 'translate the error into human-readable language' instead of 'also offer advice'. But I'm open to this later.